File: src/interfaces/address.ts
import { prop, Typegoose, ModelType, InstanceType } from 'typegoose';
import IPoint from './point';

export default class IAddress {

	@prop({ required: true })
	schema_version: number;

	@prop()
	street?: string;

	@prop()
	city?: string;

	@prop()
	province?: string;

	@prop()
	country?: string;

	@prop()
	postal_code?: string;

	@prop()
	apt_number?: string;

	@prop()
	geojson?: IPoint;
}

File: src/interfaces-internal/address.go
package interfaces_internal

type IAddress struct {
	SchemaVersion uint32 `bson:"schema_version" json:"schema_version"`
	Street        string `bson:"street" json:"street"`
	City          string `bson:"city" json:"city"`
	Province      string `bson:"province" json:"province"`
	Country       string `bson:"country" json:"country"`
	PostalCode    string `bson:"postal_code" json:"postal_code"`
	AptNumber     string `bson:"apt_number" json:"apt_number"`
	GeoJSON       IPoint `bson:"geojson" json:"geojson"`
}